etl/steps/data/meadow/ggdc/2022-11-28/penn_world_table_national_accounts.py

```diff
@@ -1,75 +1,47 @@
-from typing import cast
-
+import owid.catalog.processing as pr
 import pandas as pd
-from owid.catalog import Dataset, Table, TableMeta
-from owid.catalog.utils import underscore_table
-from owid.walden import Catalog as WaldenCatalog
 from structlog import get_logger
 
-from etl.helpers import PathFinder
-from etl.steps.data.converters import convert_walden_metadata
+from etl.helpers import PathFinder, create_dataset
 
 log = get_logger()
 
 # naming conventions
-N = PathFinder(__file__)
+paths = PathFinder(__file__)
 
 
 def run(dest_dir: str) -> None:
     log.info("penn_world_table_national_accounts.start")
 
-    # retrieve raw data from walden
-    walden_ds = WaldenCatalog().find_one(
-        namespace="ggdc", short_name="penn_world_table_national_accounts", version="2021-06-18"
-    )
-    local_file = walden_ds.ensure_downloaded()
-
-    df = pd.read_excel(local_file, sheet_name="Data")
+    # retrieve raw data
+    snap = paths.load_snapshot("penn_world_table_national_accounts.xlsx")
+    tb = snap.read_excel(sheet_name="Data")
 
     # # clean and transform data
-    # df = clean_data(df)
+    # tb = clean_data(tb)
 
     # Read reference dataset for countries and regions
-    df_countries_regions = cast(Dataset, N.load_dependency("regions"))["regions"]
+    tb_countries_regions = paths.load_dataset("regions")["regions"]
 
     # Merge dataset and country dictionary to get the name of the country (and rename it as "country")
-    df = pd.merge(
-        df, df_countries_regions[["name", "iso_alpha3"]], left_on="countrycode", right_on="iso_alpha3", how="left"
+    tb = pr.keep_metadata(pd.merge)(
+        tb, tb_countries_regions[["name", "iso_alpha3"]], left_on="countrycode", right_on="iso_alpha3", how="left"
```

Comment:
wouldn't `pr.merge` do the job here?

Reply:
It probably would, I don't remember why I used this. I'll fix it.